src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -58,14 +62,38 @@
       final String symbol,
       final String range,
       final String date) {
+    List<IexHistoricalPrice> historicalPriceList;
     if (date == null && range == null) {
-      return iexCloudClient.getHistoricalPricesSymbol(symbol);
+      historicalPriceList = iexCloudClient.getHistoricalPricesSymbol(symbol);
     } else if (range == null) {
-      return iexCloudClient.getHistoricalPricesDate(symbol, date);
+      historicalPriceList = iexCloudClient.getHistoricalPricesDate(symbol, date);
     } else if (date == null) {
-      return iexCloudClient.getHistoricalPricesRange(symbol, range);
+      historicalPriceList = iexCloudClient.getHistoricalPricesRange(symbol, range);
     } else {
-      return iexCloudClient.getHistoricalPrices(symbol, range, date);
+      historicalPriceList = iexCloudClient.getHistoricalPrices(symbol, range, date);
+    }
+
+    addHistoricalPricesToDB(historicalPriceList);
+
+    return historicalPriceList;
+  }
+
+  /**
+   * Convert each historical price to an entity and add to the database.
+   * @param historicalPriceList a list of Historical Prices fetched from the IEX API
+   */
+  public void addHistoricalPricesToDB(final List<IexHistoricalPrice> historicalPriceList) {
+    for (IexHistoricalPrice price : historicalPriceList) {
+      IexHistoricalPriceEntity entity = new IexHistoricalPriceEntity(
+          price.getSymbol(),
+          price.getDate(),
+          price.getClose(),
+          price.getHigh(),
+          price.getLow(),
+          price.getOpen(),
+          price.getVolume());
+      historicalPriceRspy.save(entity);
```

Comment:
Using saveAll would be more efficient due to the way hibernate will execute a save, but it's not too important here.